qiskit/pulse/instructions/acquire.py

```diff
@@ -113,23 +113,26 @@ def __init__(self,
             name = 'acq{:10x}'.format(hash((duration, tuple(channels), tuple(mem_slot),
                                             tuple(reg_slot), kernel, discriminator)))
 
-        if channels is not None:
-            super().__init__(duration, *channels, *mem_slot, *reg_slot, name=name)
-        else:
-            super().__init__(duration, name=name)
-
         self._acquires = channels
         self._channel = channels[0] if channels else None
         self._mem_slots = mem_slot
         self._reg_slots = reg_slot
         self._kernel = kernel
         self._discriminator = discriminator
 
+        if channels is not None:
+            super().__init__(duration, *channels, *mem_slot, *reg_slot, name=name)
+        else:
+            warnings.warn("Usage of Acquire without specifying a channel is deprecated. For "
+                          "example, Acquire(1200)(AcquireChannel(0)) should be replaced by "
+                          "Acquire(1200, AcquireChannel(0)).", DeprecationWarning)
+            super().__init__(duration, name=name)
+
     @property
-    def operands(self) -> List:
-        """Return a list of instruction operands."""
-        return [self.duration, self.channel,
-                self.mem_slot, self.reg_slot]
+    def operands(self) -> Tuple[int, AcquireChannel, MemorySlot, RegisterSlot]:
```

Comment:
I haven't seen that restriction anywhere else, what does that stem from? I've seen tests with acquire taking memslot and reg slot. No one uses register slots yet, so it's possible they're not handled correctly in a couple places.

Reply:
Can users not use readout results for fast feedback and also get that result?

Reply:
It will be a practical restriction in the hardware. In reality it would be something like
1. acquire qubit into register
2. store register into memory
Storing directly into a memory_slot is currently supported but is a convenience that masks hardware implementation. I would like to get away from multi-purpose instructions and I think this is a good first step that can be made while we are already changing the interface.

Reply:
so Acquire should really only take a register slot. This seems like an implementation change that is out of scope for this PR. Maybe next sprint?

Reply:
Will we be able to make the change before release?